### PR TITLE
fix #25143 Successive calls to `precacheImage()` throw an exception

### LINKED ISSUE #25143: Successive calls to `precacheImage()` throw an exception
## Steps to Reproduce

1. Call `precacheImage(const NetworkImage('https://via.placeholder.com/350x150'), context);`
2. Wait a few seconds, allowing for the image to cache.
3. Call the same line of code.

I found this easiest to reproduce by placing the line of code in a `IconButton`'s `onPressed` handler. Pressing the button the first time works fine, but pressing it a second time consistently produces the following error.

For added context, `package:Checkiday/main.dart:270:15` in the below logs is the line of code above.

## Logs

```
[+7803 ms] I/flutter ( 9077): ══╡ EXCEPTION CAUGHT BY IMAGE RESOURCE SERVICE ╞════════════════════════════════════════════════════
[   +8 ms] I/flutter ( 9077): The following StateError was thrown by an image error listener:
[        ] I/flutter ( 9077): Bad state: Future already completed
[  +10 ms] I/flutter ( 9077): 
[        ] I/flutter ( 9077): When the exception was thrown, this was the stack:
[  +10 ms] I/flutter ( 9077): #1      precacheImage.errorListener (package:flutter/src/widgets/image.dart:88:15)
[        ] I/flutter ( 9077): #2      ImageStreamCompleter.reportError (package:flutter/src/painting/image_stream.dart:327:24)
[        ] I/flutter ( 9077): #3      ImageStreamCompleter.setImage (package:flutter/src/painting/image_stream.dart:284:9)
[        ] I/flutter ( 9077): #4      MultiFrameImageStreamCompleter._emitFrame (package:flutter/src/painting/image_stream.dart:519:5)
[        ] I/flutter ( 9077): #5      MultiFrameImageStreamCompleter._decodeNextFrameAndSchedule (package:flutter/src/painting/image_stream.dart:512:7)
[        ] I/flutter ( 9077): <asynchronous suspension>
[        ] I/flutter ( 9077): #6      MultiFrameImageStreamCompleter.addListener (package:flutter/src/painting/image_stream.dart:528:7)
[        ] I/flutter ( 9077): #7      ImageStream.addListener (package:flutter/src/painting/image_stream.dart:156:25)
[        ] I/flutter ( 9077): #8      precacheImage (package:flutter/src/widgets/image.dart:101:10)
[        ] I/flutter ( 9077): #9      _HomePageState.build.<anonymous closure> (package:Checkiday/main.dart:270:15)
[        ] I/flutter ( 9077): #10     _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:507:14)
[        ] I/flutter ( 9077): #11     _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:562:30)
[        ] I/flutter ( 9077): #12     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
[        ] I/flutter ( 9077): #13     TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:242:9)
[        ] I/flutter ( 9077): #14     TapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:204:7)
[        ] I/flutter ( 9077): #15     GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:156:27)
[        ] I/flutter ( 9077): #16     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:184:20)
[        ] I/flutter ( 9077): #17     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:158:22)
[        ] I/flutter ( 9077): #18     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:138:7)
[        ] I/flutter ( 9077): #19     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:101:7)
[        ] I/flutter ( 9077): #20     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:85:7)
[        ] I/flutter ( 9077): #21     _invoke1 (dart:ui/hooks.dart:168:13)
[        ] I/flutter ( 9077): #22     _dispatchPointerDataPacket (dart:ui/hooks.dart:122:5)
[        ] I/flutter ( 9077): (elided one frame from package dart:async)
[        ] I/flutter ( 9077): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

```
[✓] Flutter (Channel stable, v1.0.0, on Linux, locale en_US.UTF-8)
    • Flutter version 1.0.0 at /home/westy92/Source/flutter
    • Framework revision 5391447fae (10 days ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /home/westy92/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /home/westy92/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] Android Studio (version 3.2)
    • Android Studio at /home/westy92/android-studio
    • Flutter plugin version 31.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.28.0)
    • VS Code at /usr/share/code
    • Flutter extension version 2.21.1

[✓] Connected device (1 available)
    • Pixel • FA69H0305733 • android-arm64 • Android 9 (API 28)

• No issues found!
```

### PR DESCRIPTION
**Pull request by truongsinh**
~I still can't figure how to write unit test for this...~

### CODE REVIEW

**Comment by goderbauer** [comment]
Fixes https://github.com/flutter/flutter/issues/25143. /cc @jonahwilliams

File: packages/flutter/lib/src/widgets/image.dart [hunk 1]
```diff
@@ -99,7 +101,6 @@ Future<void> precacheImage(
     }
   }
   stream.addListener(listener, onError: errorListener);
-  completer.future.then<void>((void value) { stream.removeListener(listener); });
```
[the remark this answers is not in the document]
Comment (westy92) [reply packages/flutter/lib/src/widgets/image.dart]: I'm not too familiar with the `Completer` API. Could we also `return` this line, dropping the line after?

**Comment by truongsinh** [comment]
unit test added

**Comment by westy92** [comment]
Awesome!

**Comment by truongsinh** [comment]
`tests-windows` https://github.com/flutter/flutter/pull/25159/checks?check_run_id=39791787 failed because of time out :-?

**Comment by jonahwilliams** [comment]
I restarted the windows tests

**Comment by westy92** [comment]
They all passed! 🎉

**Comment by xster** [comment]
LGTM. Thanks for the tests.

**Pull request merged**